src/stt/src/flt/flute.cpp

```diff
@@ -1008,8 +996,8 @@ Tree flute(const std::vector<DTYPE>& x, const std::vector<DTYPE>& y, int acc) {
     xs.resize(d);
     ys.resize(d);
     s.resize(d);
-    pt = (struct point *)malloc(sizeof(struct point) * (d + 1));
-    std::vector<point*> ptp(d+1);
+    pt = (struct point *)malloc(sizeof(struct point) * (d));
```

Comment:
you found both bugs, but this fix is not compatible with the code right below the sort:
#if FLUTE_REMOVE_DUPLICATE_PIN == 1
    ptp[d] = &pt[d];
    ptp[d]->x = ptp[d]->y = -999999;

IE, there is a reason the original flute code used d+1 for the array size.

Reply:
In my tests, it didn't error out because FLUTE_REMOTE_DUPLICATE_PIN is 0 by default. But I should have notice the code below anyway.

As a possible fix, I can change the stable_sort call to:
`std::stable_sort(ptp.begin(), `**`ptp.end()-1`**`, ordery);`.
Is this solution ok?

Reply:
I believe that is the correct solution